### PR TITLE
Feature/9 application entity

### PR DESCRIPTION
**Pull request by ThetaSinner**
Not ready to merge but initial implementation is done

### CODE REVIEW

File: acropolis-persistence/src/main/kotlin/org/ephyra/acropolis/persistence/api/SystemSoftwarePersistence.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.ephyra.acropolis.persistence.api.entity.SystemSoftwareEntity
+import org.ephyra.acropolis.persistence.impl.SystemSoftwareRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Component
+class SystemSoftwarePersistence {
+    @Autowired
+    private lateinit var repo: SystemSoftwareRepository
```
Comment (Sub-Xaero) [inline acropolis-persistence/src/main/kotlin/org/ephyra/acropolis/persistence/api/SystemSoftwarePersistence.kt]: Probably want to try and be consistent with naming this. Called `repo` in some places and `persistence` in others.

File: acropolis-persistence/src/main/kotlin/org/ephyra/acropolis/persistence/api/entity/ApplicationSoftwareEntity.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ *  - A word processor
+ *  - An interactive shell or terminal
+ */
+@Entity
+class ApplicationSoftwareEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    var id: Int = 0
+
+    var name: String = ""
```
Comment (Sub-Xaero) [inline acropolis-persistence/src/main/kotlin/org/ephyra/acropolis/persistence/api/entity/ApplicationSoftwareEntity.kt]: Do we want to introduce any validations? `@NotNull`?

Comment (ThetaSinner) [reply acropolis-persistence/src/main/kotlin/org/ephyra/acropolis/persistence/api/entity/ApplicationSoftwareEntity.kt]: Yes absolutely, but I don't know the javax.persistence API so one of the things to do before this PR is ready is go through some documentation and get the persistence tidied up into a nice example for future work

Comment (ThetaSinner) [reply acropolis-persistence/src/main/kotlin/org/ephyra/acropolis/persistence/api/entity/ApplicationSoftwareEntity.kt]: Yeah there are annotations that can help describe the data model to the framework. It's the standard javax.persistence API that's implemented by Hibernate and wrapped up in Springy stuff. I saw you picked up 'datastore' so if you get to this before I do then that should be the place to look :)

**Comment by ThetaSinner** [comment]
Pretty happy with this. There's lots more that could be done but want to stay focused on the task.

Feel free to review and merge when ready unless there are any more comments, then you'll be able to use the changes. Hopefully I haven't created too terrible a merge!

**Pull request merged**